File: app/src/main/java/com/mehmetakiftutuncu/muezzin/utilities/LocaleUtils.java
package com.mehmetakiftutuncu.muezzin.utilities;

import android.content.Context;
import android.support.annotation.NonNull;

import java.text.Collator;
import java.util.Locale;

/**
 * Created by akif on 11/05/16.
 */
public class LocaleUtils {
    public static boolean isLanguageTurkish(Context context) {
        return getCurrentLanguage(context).contains("tr");
    }

    public static boolean isLanguageEnglish(Context context) {
        return getCurrentLanguage(context).contains("en");
    }

    public static Collator getCollator(@NonNull Context context) {
        return isLanguageTurkish(context) ? Collator.getInstance(new Locale("tr", "TR")) : (isLanguageEnglish(context) ? Collator.getInstance(Locale.ENGLISH) : Collator.getInstance());
    }

    public static Collator getTurkishCollator() {
        return Collator.getInstance(new Locale("tr", "TR"));
    }

    private static String getCurrentLanguage(Context context) {
        return context.getResources().getConfiguration().locale.getLanguage();
    }
}

File: app/src/main/java/com/mehmetakiftutuncu/muezzin/utilities/LocaleUtils.kt
package com.mehmetakiftutuncu.muezzin.utilities

import android.content.Context
import com.mehmetakiftutuncu.muezzin.models.Country
import java.text.Collator
import java.util.*

object LocaleUtils {
    val turkishCollator: Collator = Collator.getInstance(Locale("tr", "TR"))

    fun isLanguageTurkish(ctx: Context): Boolean =
        getCurrentLanguage(ctx).contains("tr")

    fun isLanguageEnglish(ctx: Context): Boolean =
        getCurrentLanguage(ctx).contains("en")


    fun getCollator(ctx: Context): Collator =
        when {
            isLanguageTurkish(ctx) -> turkishCollator
            isLanguageEnglish(ctx) -> Collator.getInstance(Locale.ENGLISH)
            else                       -> Collator.getInstance()
        }

    fun List<Country>.turkeyFirstSorted(ctx: Context): List<Country> {
        val collator  = getCollator(ctx)
        val isTurkish = isLanguageTurkish(ctx)
        val isEnglish = isLanguageEnglish(ctx)

        val (turkey, others) = this.sortedWith { c1, c2 ->
            collator.compare(
                if (isTurkish) c1.nameTurkish else if (isEnglish) c1.name else c1.nameNative,
                if (isTurkish) c2.nameTurkish else if (isEnglish) c2.name else c2.nameNative
            )
        }.partition { it.isTurkey }

        return turkey + others
    }

    private fun getCurrentLanguage(ctx: Context): String =
        if (android.os.Build.VERSION.SDK_INT < android.os.Build.VERSION_CODES.N) {
            ctx.resources.configuration.locale.language
        } else {
            ctx.resources.configuration.locales[0].language
        }
}